src/arrayMethodJoin.js

```diff
@@ -4,8 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const addition = String(separator);
+    let newStr = '';
+
+    if (this.length === 0) {
+      return newStr;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      switch (this[i]) {
+        case null:
+          newStr += '' + addition;
+          break;
+        case undefined:
+          newStr += '' + addition;
+          break;
+        default:
+          newStr += this[i] + addition;
+          break;
+      }
+    }
+
+    if (this[this.length - 1] !== null
+    && this[this.length - 1] !== undefined) {
+      newStr += this[this.length - 1];
+    }
```

Comment:
you check this if right now for each cycle iteration
let's think how we can rewrite logic for better separator processing